src/Features/Core/Portable/Completion/CharacterSetModificationRule.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    /// <summary>
+    /// A rule that modifies a set of characters.
+    /// </summary>
+    public sealed class CharacterSetModificationRule
+    {
+        /// <summary>
+        /// The kind of modification.
+        /// </summary>
+        public CharacterSetModificationKind Kind { get; }
+
+        /// <summary>
+        /// One or more characters.
+        /// </summary>
+        public string Characters { get; }
```

Comment:
I feel like this being a string in the public API isn't appropriate.  Feels like it should be an ImmutableArray<char>.  Internally, if you want to store it as a string in things like the controller, that seems fine.